fastify.js

```diff
@@ -85,15 +86,15 @@ function build (options) {
   })
 
   const requestIdHeader = options.requestIdHeader || 'request-id'
+  const genReqId = options.genReqId || reqIdGenFactory(requestIdHeader)
 
   fastify.printRoutes = router.prettyPrint.bind(router)
 
   const setupResponseListeners = Reply.setupResponseListeners
+
   // logger utils
-  const customGenReqId = options.logger ? options.logger.genReqId : null
```

Comment:
Can you throw if genReqId is specified under logger? Or move it to the main option object and emit a warning (process.emitWarning)?

Reply:
#1166